src/test/scala/nl.knaw.dans.easy.deposit/authentication/SessionSpec.scala

```diff
@@ -148,29 +149,30 @@ class SessionSpec extends TestSupportFixture with ServletFixture with ScalatraSu
       uri = "/auth/login",
       headers = Seq(("Authorization", fooBarBasicAuthHeader))
     ) {
-      body shouldBe "signed in"
-      status shouldBe OK_200
-      header("Content-Type") shouldBe "text/plain;charset=UTF-8"
+      body shouldBe ""
+      status shouldBe NO_CONTENT_204
+      header("Content-Type") shouldBe "text/html;charset=UTF-8"
       header("Expires") shouldBe "Thu, 01 Jan 1970 00:00:00 GMT" // page cache
       header("REMOTE_USER") shouldBe "foo"
       val newCookie = header("Set-Cookie")
       newCookie should startWith("scentry.auth.default.user=")
+      newCookie shouldNot startWith("scentry.auth.default.user=;")
       newCookie should include(";Path=/")
       newCookie should include(";HttpOnly")
       cookieAge(newCookie) should be < 1000L
     }
   }
 
-  "put /auth/logout" should "clear the cookie" in {
+  "post /auth/logout" should "clear the cookie" in {
```

Comment:
mmm, is this testing the `AuthServlet` or the `AuthTestServlet`? I would expect the former...

Reply:
this is testing `AuthTestServlet`

Reply:
The question is exactly the point of the TODO above. When I initially wrote it, I did not fight long enough to find a solution. It is beyond the scope of the issue, therefore I recommended a TODO.

Reply:
@jo-pol : can you comment on why the separate `TestServlet`-classes are necessary?

Reply:
Any servlet can prove a log-in succeeded. With a TestServlet we are looser coupled.

The story behind TestAuthServlet is more complex. The constructor of AuthServlet requires EasyDepositApiApp for which we should provide a mocked AuthenticationProvider. This test was written as a proof of concept but wasn't changed into something that tested the actual class.

A quick attempt to fix it (shown below) failed. Let us keep it out of this PR.

```
  addServlet(new AuthServlet(new EasyDepositApiApp(minimalAppConfig)) {
    override def getAuthenticationProvider: AuthenticationProvider = mockedAuthenticationProvider
  }, path = "/auth/*")
```